theories/encryption/Hybrid.ec

```diff
@@ -540,4 +648,12 @@ section.
   + by conseq (: _ ==> ={res, glob Ob})=> //; sim.
   by auto=> /> l0 /supp_dinter /#.
   qed.
+  
+  lemma Hybrid_restr_div &m (p:glob A -> glob Ob -> int -> outputA -> bool):
+      q <> 0 =>
+        Pr[HybGame(A,Ob,L(Ob)).main() @ &m : p (glob A) (glob Ob) HybOrcl.l res]
+      - Pr[HybGame(A,Ob,R(Ob)).main() @ &m : p (glob A) (glob Ob) HybOrcl.l res]
+    = 1%r/q%r * (  Pr[Ln(Ob,A).main() @ &m : p (glob A) (glob Ob) Count.c res]
+                 - Pr[Rn(Ob,A).main() @ &m : p (glob A) (glob Ob) Count.c res]).
+  proof. by move=> qN0; rewrite Hybrid_restr; smt(q_ge0). qed.
```

Comment:
It's worth thinking about whether this lemma is generally useful in this form: We're usually trying to upper-bound the advantage of `A` in distinguishing `Ln` from `Rn`, so the multiplicative form is probably the most useful?

Reply:
Yes, I agree that the multiplicative from is the one that's most useful. However, the form with division was previously the only available form, so having these lemmas eases porting.